### ISSUE TITLE
Prep for Cloud Next

### ISSUE THREAD

**Issue by briand44** [uiowa/uiowa#7263]
# Story

As a web team we would like our applications to be ready for the upgrade to Cloud Next so we don't experience any issues when these are upgraded.

# Additional Info

Issue to allowlist web EIPs https://github.com/uiowa/uiowa/issues/7252

Here is the last ticket that shows potential issues affection each applications upgrade to Cloud Next.

https://acquia.my.site.com/s/case/5006g00000r0c1PAAQ/acquia-cloud-next-acn-upgrade-eligibility

**Pending Customer Changes**

uiowa701 ([WebEIPs](https://docs.acquia.com/cloud-platform/manage/domains/web-eip/), Log Fwding Cert**)
uiowa702 ([WebEIPs](https://docs.acquia.com/cloud-platform/manage/domains/web-eip/), [Cron](https://docs.acquia.com/cloud-platform/upgrade-acn/customer-changes/#cron), Log Fwding Cert**)

**Currently Upgrading - Pending Web EIP Confirmation Only**

uiowa06 ([WebEIPs](https://docs.acquia.com/cloud-platform/manage/domains/web-eip/))
uiowa05 ([WebEIPs](https://docs.acquia.com/cloud-platform/manage/domains/web-eip/))
uiowa703 ([WebEIPs](https://docs.acquia.com/cloud-platform/manage/domains/web-eip/), [SAML](https://docs.acquia.com/cloud-platform/upgrade-acn/customer-changes/#saml), [Cron](https://docs.acquia.com/cloud-platform/upgrade-acn/customer-changes/#cron))
uiowa03 ([WebEIPs](https://docs.acquia.com/cloud-platform/manage/domains/web-eip/))
uiowa04 ([WebEIPs](https://docs.acquia.com/cloud-platform/manage/domains/web-eip/))
uiowa01 ([WebEIPs](https://docs.acquia.com/cloud-platform/manage/domains/web-eip/))
uiowa02 ([WebEIPs](https://docs.acquia.com/cloud-platform/manage/domains/web-eip/))
uigraduatecollege ([WebEIPs](https://docs.acquia.com/cloud-platform/manage/domains/web-eip/))
uiclas ([WebEIPs](https://docs.acquia.com/cloud-platform/manage/domains/web-eip/))
uidentistry ([WebEIPs](https://docs.acquia.com/cloud-platform/manage/domains/web-eip/), [SAML](https://docs.acquia.com/cloud-platform/upgrade-acn/customer-changes/#saml))

**Future Deliverables**

uiowa - Unsupported PHP Extension ([WebEIPs](https://docs.acquia.com/cloud-platform/manage/domains/web-eip/))

# Proposed solution
```[tasklist]
### Tasks
- [x] Update log forwarding cert on 701 and 702
- [x] Any changes to cron? (702, 703)
- [x] Address unsupported PHP extension on uiowa - https://github.com/uiowa/uiowa/issues/7263#issuecomment-1883254894
```

**Comment by briand44** [comment]
>Regarding the unsupported php extension, it looks like this is tied to "Xdebug" which you have in your dev environments in both those applications. It looks like this will eventually be supported, but not until later this year (Q3). The immediate option would be to see if this is needed / could be removed, or we look to find a different supported option. I am working with the ACN team on alternative options too.

**Comment by briand44** [comment]
This is the schedule for non-production environment upgrades based on the Acquia tickets. Brittany said she would request to delay upgrades for another few weeks but the Acquia tickets don't represent that delay currently so I'm not super confident.

~uiowa05 - Jan 11th - Jan 12th 10pm - 6am **postponed** now Feb 5th 10pm **postponed** now Feb 14th 10pm~
~uiowa06 - Jan 11th - Jan 12th 10pm - 6am **postponed** now Feb 5th 10pm **postponed** now Feb 14th 10pm~
~uiowa703 - Jan 11th - Jan 12th 10pm - 6am **postponed** now Feb 5th 10pm **postponed** now Feb 14th 10pm~
~uiowa03 - Jan 15th - Jan 16th 10pm - 6am **postponed** now Feb 7th 10pm~ upgraded
~grad - Jan 15th - Jan 16th 10pm - 6am~ upgraded
~clas - Jan 15th - Jan 16th 10pm - 6am~ upgraded
~uiowa04 - Jan 16th - Jan 17th 10pm - 6am~ upgraded
~Dentistry - Jan 16th - Jan 17th 10pm - 6am~ upgraded
~uiowa702 - Jan 16th - Jan 17th 10pm - 6am~ upgraded
~uiowa701 - Jan 16th - Jan 17th 10pm - 6am **postponed** now Feb 12th 10pm~
~uiowa01 - Jan 17th - Jan 18th 10pm - 6am~ upgraded?? - Looks like upgraded but got email saying postponed?
~uiowa02 - Jan 18th - Jan 19th 10pm - 6am~ upgraded

**Not Scheduled**
uiowa

**Done**
Signage
HC Signage
uiowa07

**Comment by briand44** [comment]
Grooming conversation... Cron needs to be minimum of 5 minutes for Cloud Next. We have some running every minute. One of which is related to Hawk Alerts. We can try and get clarification in our meeting tomorrow with Emergency stakeholders. We didn't update SAML for signage and authentication still works. Log forwarding cert, we should email Alex and keep Nikki cc'd. There is a thread in its-web with their contact info.

**Comment by joewhitsitt** [comment]
Just to note, we had to adjust some things related to drush when setting up uiowa07:

https://github.com/uiowa/uiowa/pull/7148, https://github.com/uiowa/uiowa/pull/7149

**Comment by joewhitsitt** [comment]
Regarding cron on 702. We need that for up to the minute updates from Rave to emergency.uiowa.edu. Mentioned to Brian that we could modify and turn on uiowa_alerts for d7 to pull from rave directly if the emergency site to maintain quick 30 second pulls of data outside of cron until the rave content can be pulled in as db content.

Otherwise, we are probably looking at bigger JS changes to make sure the site doesn't say "Campus Status Normal" for 5 min when in fact things are not. At that point, maybe we just prioritize getting them on v3 and accounting for this cron job delay there.

**Comment by briand44** [comment]
Is the wrapper script mentioned here https://docs.acquia.com/cloud-platform/upgrade-acn/customer-changes/#cron an option?

**Comment by joewhitsitt** [comment]
🤷

**Comment by briand44** [comment]
Response from Acquia on my questions related to SAML.

>you no longer need to do the code changes to move up to ACN in this case. UIowa upgraded some apps to ACN that use SimpleSAMLphp without making the code changes our docs say are required. Are there cases where this is possible? Yes, we made this a platform change in Q3 of last year. However, it’s still a good idea if they can get to the newer code as it’s more efficient, but it’s no longer a hard blocker. The technical details are that we added a hand built Class for the Net::DNS2 which mimics the outcome of how it works on Classic.

**Comment by briand44** [comment]
Related to cron.

>The cron frequency change is not in the ACN roadmap at this point. Some options:
Best thing to do is essentially client-side rendering, and a 3rd party (non-Drupal) API endpoint.
We solved the notifications issue at University of Nebraska-Lincoln with JS and a third-party notification system:
https://github.com/unl/wdntemplates/blob/5.3/wdn/templates_5.3/includes/global/header-global-1.html 
https://github.com/unl/wdntemplates/blob/5.3/wdn/templates_5.3/js-src/banner.babel.js 
It makes a call to https://alert.unl.edu/json/unlcap.js every 30 seconds. That system is run by a third party.
And also, as you already know, there is a wrapper script that can be run in ACN to run a CRON every minute:
https://docs.acquia.com/cloud-platform/upgrade-acn/customer-changes/#reduce-the-frequency-of-cron-jobs

**Comment by joewhitsitt** [comment]
I think I understand the "wrapper script" a bit more. We would define a scheduled job for 5 min that calls the script and the script runs until 5 min passes. https://docs.acquia.com/cloud-platform/manage/cron/advanced-cron/#running-a-shell-script

It would have been helpful if they had linked to this ^^ from https://docs.acquia.com/cloud-platform/upgrade-acn/customer-changes/#reduce-the-frequency-of-cron-jobs

**Comment by joewhitsitt** [comment]
Planning on picking this up later today if nobody else is interested.

Update: Sean expressed interest! Yay!

**Comment by joewhitsitt** [comment]
>Update log forwarding cert on 701 and 702

Alex responded to the EIPs email pretty quickly. @pyrello have you sent a similar message yet for these certs?

**Comment by pyrello** [comment]
> > Update log forwarding cert on 701 and 702
> 
> Alex responded to the EIPs email pretty quickly. @pyrello have you sent a similar message yet for these certs?

I have not. It wasn't totally clear to me what the next step was here.

**Comment by joewhitsitt** [comment]
ISPO (Alex Bennett, formally Nikki) has permissions to manage the log forwarding certs on our applications, but they don't get notified that they expire.

**Comment by pyrello** [comment]
Sent an email to Alex Bennett in ISPO requesting to update the log forwarding certs for 701 and 702.

**Comment by pyrello** [comment]
@briand44 I'm wondering if there is anything to do related to SAML?

**Comment by briand44** [comment]
@pyrello Nothing required to do for SAML, see https://github.com/uiowa/uiowa/issues/7263#issuecomment-1896478358. Sounds like new code would be more efficient but not a blocker for moving.

**Comment by pyrello** [comment]
> @pyrello Nothing required to do for SAML, see [#7263 (comment)](https://github.com/uiowa/uiowa/issues/7263#issuecomment-1896478358). Sounds like new code would be more efficient but not a blocker for moving.

I saw that comment, but was confused since "Any changes needed for SAML? (Dentistry, 703,)" was still listed in the proposed solution.

**Comment by pyrello** [comment]
I opened a ticket with Acquia support to turn off Xdebug on the dev environment for `uiowa`.

**Comment by pyrello** [comment]
This is a frustratingly small amount of information about cron wrapper scripts: https://docs.acquia.com/cloud-platform/upgrade-acn/customer-changes/#reduce-the-frequency-of-cron-jobs

**Comment by pyrello** [comment]
> I opened a ticket with Acquia support to turn off Xdebug on the dev environment for `uiowa`.

This has been resolved.

**Comment by pyrello** [comment]
I'm having trouble with permissions. I tried making the file executable, but that doesn't seem to have worked. I tried following these instructions: https://acquia.my.site.com/s/article/1500000899062-Adding-an-executable-script-and-its-folder-to-a-site-s-Git-repository. No luck. Going to open a support ticket.

**Comment by joewhitsitt** [comment]
Based on https://github.com/uiowa/uiowa/issues/7224#issuecomment-1919517895 for visibility, I have removed the  `inrc.uiowa.edu safe-cache-form-clear` scheduled job as it should no longer be needed. There is a separate hourly job for this site that should remain for now but is not a blocker for ACN.

**Comment by pyrello** [comment]
I got very close to getting the cron swapped out today. There seems to be an issue with the drush command not being found when the new script runs in prod. I don't recall seeing this issue in dev. Next step is to troubleshoot that.

**Comment by joewhitsitt** [comment]
>There seems to be an issue with the drush command not being found when the new script runs in prod

Not sure if this is the case, but have you tried adding the relative path to drush? I know the doc example above doesn't have it, but it seems like everything else related to these jobs needs the path prefix.

`/usr/local/bin/drush`

**Comment by pyrello** [comment]
> > There seems to be an issue with the drush command not being found when the new script runs in prod
> 
> Not sure if this is the case, but have you tried adding the relative path to drush? I know the doc example above doesn't have it, but it seems like everything else related to these jobs needs the path prefix.
> 
> `/usr/local/bin/drush`

Sorry, my language was a bit unclear. `drush` works, but its not finding the specific drush command. I think its not in the correct context to know which site to load.

@bspeare and I did a little bit of poking at this earlier and were able to figure out how to resolve some environmental variables so that we could add the alias for the application back in and that it would work for each environment.

**Comment by pyrello** [comment]
Today:
- Got a version of the shell script working that uses environment variables to construct the alias dynamically and therefore doesn't need to be modified for testing in a non-prod environment.
- Tested it in production and validated that runs on the same cadence as the existing job with @joewhitsitt and @briand44.
- Contacted a couple members of the Public Safety team about create a test alert, which sounds like a much more involved process than I was expecting.

**Comment by pyrello** [comment]
We ran a test this morning propagating a Hawk Alert to the dev Emergency site and I monitored the logs on 702 dev. Everything went as expected.

Next step is to try turning off the old scheduled job for a while and monitor logs to see if there is anything concerning when only one job is running. If no issues there, then I think this will be completed.

**Comment by pyrello** [comment]
We're on the new scheduled job using the cron wrapper script. So, I think this is done.
